### ISSUE TITLE
[MQTT] Need to reintroduce credentials to MQTT broker when container is self-updated

### ISSUE THREAD

**Issue by gboulvin** [GladysAssistant/Gladys#1368]
Caution ! It might be relative to eclipse-mosquito container, not directly to Gladys.
Actually, every time the eclipse-mosquito container is updated, user needs to reintroduce the password in Gladys's configuration page and click the 'save' button so it, and other peripherals, can get connected again.

**Comment by atrovato** [comment]
What do you means with "every time the eclipse-mosquito container is updated"?
Docker image is upgraded? Configuration changed from Gladys?

**Comment by VonOx** [comment]
I've never experienced this behavior

**Comment by LM1LC3N7** [comment]
How did you deployed Gladys and eclipse-mosquito? Did you use Gladys directly or manually deployed the mosquito container?

**Comment by gboulvin** [comment]
Gladys was installed with the official image.
It happens when the eclipse-mosquitto container gets an upgrade (I guess it's watchtower's job).
It acts like if the new version of the container looses its previous configuration.
We've discussed about this here https://community.gladysassistant.com/t/resolu-probleme-de-connexion-mqtt-avec-sonoff/6095/27?u=gboulvin

**Comment by Pierre-Gilles** [comment]
@atrovato @VonOx I think the issue is that Gladys is not "auto-reconnecting" to the MQTT broker when the broker restarts, it's not related to credentials

**Comment by gboulvin** [comment]
But even the devices (Tasmota, customs) can't connect to the broker again as if it was offline...

**Comment by VonOx** [comment]
@Pierre-Gilles not really as @gboulvin describe no device can connect to the broker. We need more information on the mqtt container, when problem occurs, Docker logs .

**Comment by atrovato** [comment]
I'll try to reproduce it (restart container), maybe the weekend...

**Comment by atrovato** [comment]
Not tried yet. Not stale.

**Comment by gboulvin** [comment]
For reference, eclipse-mosquito just got updated and... Bug still exists, needed to reintroduce credentials within Gladys's configuration page.
I might try to fully reinstall instance...

**Comment by gboulvin** [comment]
Still open

Le mar. 19 juil. 2022 à 13:44, stale[bot] ***@***.***> a
écrit :

**Comment by gboulvin** [comment]
Container has been updated yesterday.
Watchtowers log:
`
time="2022-07-20T21:54:43Z" level=info msg="Found new eclipse-mosquitto:2 image (f963d242e559)"
time="2022-07-20T21:54:44Z" level=info msg="Stopping /eclipse-mosquitto (9a07b0ba4086) with SIGTERM"
time="2022-07-20T21:54:46Z" level=info msg="Creating /eclipse-mosquitto"
time="2022-07-20T21:54:46Z" level=info msg="Removing image 922c6b098fad"
time="2022-07-20T21:54:46Z" level=info msg="Session done" Failed=0 Scanned=4 Updated=1 notify=no
`
Gladys's log:
`
2022-07-20T21:54:45+0000 <warn> connect.js:50 (Mqttclient.<anonymous>) Error while connecting to MQTT - Error: write ECONNRESET
`
All devices were reconnected successfully by themselves. Gladys did not reconnect until I reintroduced the password in Gladys's Mqtt config field.
